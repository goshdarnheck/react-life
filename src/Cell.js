import React from 'react';
import PropTypes from 'prop-types';

import {
    CELL_EMPTY,
    CELL_ALIVE
} from './constants';

function Cell(props) {
    let style = {
        backgroundColor: props.state === CELL_ALIVE ? 'yellowgreen' : 'darkred',
        color: props.state ? 'white' : 'darkred',
        width: 1 * props.zoom,
        height: 1 * props.zoom,
        lineHeight: 1 * props.zoom + 'px',
        top: 0 + props.y * props.zoom,
        left: 0 + props.x * props.zoom,
    };

    return <div onClick={() => props.handleCellClick(props.x, props.y)} className="cell" data-x={props.x} data-y={props.y} style={style}>â€¢</div>;
}

Cell.propTypes = {
    zoom: PropTypes.number.isRequired,
    handleCellClick: PropTypes.func.isRequired
};

export default Cell;