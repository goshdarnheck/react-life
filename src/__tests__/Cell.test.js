import Cell from "../components/cell";
import renderer from "react-test-renderer";

test("Cell renders correctly when not alive", () => {
  const component = renderer.create(
    <Cell
      alive={false}
      hue={0}
      cellKey={"0|0"}
      x={0}
      y={0}
      handleCellClick={() => {}}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Cell renders correctly when alive", () => {
  const component = renderer.create(
    <Cell
      alive={true}
      hue={0}
      cellKey={"0|0"}
      x={0}
      y={0}
      handleCellClick={() => {}}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
