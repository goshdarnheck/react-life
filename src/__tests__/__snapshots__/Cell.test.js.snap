// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders correctly when alive 1`] = `
<button
  aria-label="0, 0: alive"
  className="cell alive"
  onClick={[Function]}
  onMouseEnter={[Function]}
  style={
    Object {
      "backgroundColor": "hsl(0, 100%, 50%)",
    }
  }
  title="0, 0"
/>
`;

exports[`Cell renders correctly when not alive 1`] = `
<button
  aria-label="0, 0: dead"
  className="cell dead"
  onClick={[Function]}
  onMouseEnter={[Function]}
  style={
    Object {
      "backgroundColor": "#222",
    }
  }
  title="0, 0"
/>
`;
